src/Client.ts

```diff
@@ -69,6 +71,15 @@ export interface ClientEvents<PeerMetadata, TrackMetadata> {
   /** Emitted when the connection is closed */
   disconnected: (client: ClientApi<PeerMetadata, TrackMetadata>) => void;
 
+  /** Emitted when on successful reconnection */
+  reconnected: (client: ClientApi<PeerMetadata, TrackMetadata>) => void;
```

Comment:
Exposing events allows me to simplify a lot of logic in Videoroom; I removed a few useEffect hooks because of that.

And you are right that it's not the React way, so I've introduced the useReconnection hook with these fields if someone needs only the current state. So, events for state changes and a hook for the current state.

```
export type UseReconnection = {
  status: ReconnectionStatus;
  isReconnecting: boolean;
  isError: boolean;
  isIdle: boolean;
};
```